### PR TITLE
implement Watir::Capabilities

### PR DESCRIPTION
**Pull request by titusfortner**
Whew. Almost a year to actually implement #436 

The important points:

1. Remove the need to create Selenium classes to initialize `Watir::Browser`; this means Watir users create hashes for all of the things and Watir plugs them into the correct objects for Selenium to use.
2. Remove the need to specify ":remote" as a browser option, just pass in browser you want, along with the server url

### CODE REVIEW

File: lib/watir/capabilities.rb [hunk 1]
```diff
@@ -0,0 +1,85 @@
+module Watir
+  class Capabilities
```
Comment (p0deje) [inline lib/watir/capabilities.rb:2]: It would be great to have some docs and examples for this.

File: lib/watirspec/implementation.rb [hunk 2]
```diff
@@ -8,6 +8,11 @@ def initialize
       @guard_proc = nil
     end
 
+    def initialize_copy(orig)
+      super
+      @browser_args = Marshal.load( Marshal.dump(@browser_args))
```
Comment (p0deje) [inline lib/watirspec/implementation.rb:13]: Unnecessary space after `(`.

Comment (p0deje) [reply lib/watirspec/implementation.rb:13]: Why you need to marshal? Why not just `#clone` or `#dup`.

**Review by p0deje — COMMENTED** [review]
Overall it's good. Just keep in mind we planned to rework capabilities in the future releases of selenium-webdriver.

**Review by p0deje — APPROVED** [review]

**Comment by titusfortner** [comment]
I held off on working on this in Watir because I thought selenium updates should be done first. Except I got burned out on the selenium work. I recognize that this will keep the Watir and selenium versions more tightly coupled for a while.

File: lib/watirspec/implementation.rb (hunk 2 above)
Comment (titusfortner) [reply lib/watirspec/implementation.rb:13]: This was a pain to figure out. A clone or dup is a different object referencing the same objects inside the browser_args array. I was really struggling especially when I was trying to do this in the specs themselves. Now that it is implemented inside the class with `#initialize_copy`, I see that I can update it to be: `@browser_args = browser_args.map(&:dup)`. Thanks for making me check again, this is much better than Marshal.

**Pull request merged**